src/main/java/nl/workingtalent/wtacademy/loan/LoanController.java

```diff
@@ -1,12 +1,131 @@
 package nl.workingtalent.wtacademy.loan;
 
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import nl.workingtalent.wtacademy.dto.ResponseDto;
+import nl.workingtalent.wtacademy.user.User;
+import nl.workingtalent.wtacademy.user.UserService;
+
 @RestController
+@CrossOrigin(maxAge = 3600)
 public class LoanController {
-	
+
 	@Autowired
 	private LoanService service;
 
+	@Autowired
+	private UserService userService;
+
+	// READ
+	@RequestMapping("loan/all")
+	public ResponseDto findAllLoans() {
+		List<Loan> loans = service.findAllLoans();
+
+		if (loans.isEmpty()) {
+			return new ResponseDto(false, null, null, "No loans found.");
+		}
+
+		Stream<ReadLoanDto> readLoanDtoStream = getLoans(loans);
+		return new ResponseDto(true, readLoanDtoStream, null,
+				loans.size() + (loans.size() < 2 ? " loan " : " loans ") + "found.");
+	}
+
+	@RequestMapping("loan/{id}")
+	public ResponseDto findLoanById(@PathVariable("id") long id) {
+		Optional<Loan> loanOptional = service.findLoanById(id);
+
+		if (loanOptional.isPresent()) {
+			Loan loan = loanOptional.get();
+			ReadLoanDto readLoanDto = new ReadLoanDto(loan);
+			return new ResponseDto(true, readLoanDto, null, "Loan found.");
+		}
+		return new ResponseDto(false, null, null, "No loan found.");
+	}
+
+	@RequestMapping("loan/search")
+	public ResponseDto searchLoan(@RequestBody SearchLoanDto dto) {
+		List<Loan> loans = service.searchLoans(dto);
+		Stream<ReadLoanDto> dtos = loans.stream().map((loan) -> {
+			return new ReadLoanDto(loan);
+		});
+		return new ResponseDto(true, dtos, null, loans.size() + (loans.size() < 2 ? " loan " : " loans ") + "found.");
+	}
+
```

Comment:
Een manier om alle loans voor een user te vinden zou handig zijn, kan via search of een aparte functie binnen LoanService